packages/task/src/browser/task-schema-updater.ts

```diff
@@ -0,0 +1,118 @@
+/********************************************************************************
+ * Copyright (C) 2019 Red Hat, Inc. and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+import { injectable, inject } from 'inversify';
+import { JsonSchemaStore } from '@theia/core/lib/browser/json-schema-store';
+import { InMemoryResources, deepClone } from '@theia/core/lib/common';
+import { IJSONSchema } from '@theia/core/lib/common/json-schema';
+import URI from '@theia/core/lib/common/uri';
+import { TaskService } from './task-service';
+
+@injectable()
+export class TaskSchemaUpdater {
+    @inject(JsonSchemaStore)
+    protected readonly jsonSchemaStore: JsonSchemaStore;
+
+    @inject(InMemoryResources)
+    protected readonly inmemoryResources: InMemoryResources;
+
+    @inject(TaskService)
+    protected readonly taskService: TaskService;
+
+    async update(): Promise<void> {
+
+        const taskSchema = {
+            properties: {
+                tasks: {
+                    type: 'array',
+                    items: {
+                        ...deepClone(taskConfigurationSchema)
+                    }
+                }
+            }
+        };
+        const taskTypes = await this.taskService.getRegisteredTaskTypes();
```

Comment:
Could it be that task types are changed dynamically? For example by an activation of a new VS Code extension. In this case the schema should be updated.

Reply:
Runners is a concept that is unique to Theia. I do not think there is a way to update them dynamically.

Reply:
ok